### PR TITLE
Beat/payment & Check Session Id

### PR DESCRIPTION
**Pull request by bagasteguhimani**
## Type of changes

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Enhancement
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behaviour?

No payment transaction for face payment
No method to check if balance has reach the limit

## What is the new behavior?

Add payment for face payment transaction
Add method to check if balance has reach the limit

## Checklist

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have filled the `[Unreleased]` [Changelog](https://github.com/nodefluxio/weber/blob/main/CHANGELOG.md

## Security Pentester API Checklist
- [ ] IDOR
- [ ] Malicious Script Injection
- [ ] SQL Injection

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

check limit
limit

payment
pay

Check Session Id
session_id

### CODE REVIEW

File: backend/controllers/payment.go [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+			return
+		}
+	}
+
+	CheckSessionResult.IsActive = facePayments.IsActive
+	if facePayments.ID != 0 {
+		CheckSessionResult.IsRegistered = true
+	} else {
+		CheckSessionResult.IsRegistered = false
+	}
```
Comment (lazuardyk) [inline backend/controllers/payment.go:186]: ini bisa langsung jadi ```CheckSessionResult.IsRegistered = true``` tanpa ngecek kondisi

File: backend/controllers/payment.go [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+func (ctrl *Controller) CheckSessionId(ctx *gin.Context) {
+	var facePayments models.FacePaymentAccount
+	var CheckSessionResult models.CheckSessionResult
+	sessionId := ctx.Param("session_id")
+
+	if err := ctrl.DBConn.Model(facePayments).First(&facePayments, "session_id = ?", sessionId).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			ctx.JSON(http.StatusNotFound, gin.H{
+				"ok":      false,
+				"message": "Session Id not found",
```
Comment (lazuardyk) [inline backend/controllers/payment.go:175]: ini kan konteksnya face payment account tapi query berdasarkan session id, kayanya response errornya perlu di rubah ```"face payment account is not found"```, bukan ```"session id not found"```. penamaan fungsi dan modelsnya juga kalo bisa jangan CheckSessionId karena ambigu, jadi CheckFacePaymentAccount aja

**Review by lazuardyk — APPROVED** [review]
nice, kalo keburu buat diedit, diedit dulu yak terkait yg gua komen. tp kalo ngga ya gpp merge aja

File: backend/controllers/payment.go (hunk 2 above)
Comment (bagasteguhimani) [reply backend/controllers/payment.go]: done https://github.com/nodefluxio/weber/pull/117/commits/2d8b3aa6c0bb78c366ec4ae75eb0e0f2ca472ce0

File: backend/controllers/payment.go (hunk 1 above)
Comment (bagasteguhimani) [reply backend/controllers/payment.go]: done https://github.com/nodefluxio/weber/pull/117/commits/2d8b3aa6c0bb78c366ec4ae75eb0e0f2ca472ce0

**Review by sonadztux — COMMENTED** [review]
Jadiin satu file aja di face_payment.go gimana? @bagasteguhimani @lazuardyk

**Comment by bagasteguhimani** [comment]
> Jadiin satu file aja di face_payment.go gimana? @bagasteguhimani @lazuardyk

Ohiya, setuju sih @sonadztux

**Comment by rizkybaihaqy** [comment]
gw dapet response kyk gini.
```
data: {
    balance: 100000
    full_name: "Elmo Turner"
    is_limit: false
}
message: "Check Limit Minimum Payment Success"
ok: false
```
gw bingung ini salah apa, kyknya ada pesan yang belum di ganti deh

**Comment by lazuardyk** [comment]
jgn di merge dulu ya. banyak yg harus difix

**Comment by lazuardyk** [comment]
Gua udah fix buat endpoint yang ```/pay```.
Beberapa yg gua fix
- Ngecek muka pake face match with enrollment, tadinya gada
- Ngecek pin kalo melebihin minimum payment doang
- Ngecek saldonya cukup ngga pake balance, tadinya ngecek malah make minimum payment
- Ngurangin balance setelah transaksi, tadinya malah ngurangin minimum payment
- Minor changes terkait penamaan fungsi & variable

**Comment by bagasteguhimani** [comment]
Thankyou Lazu @lazuardyk, Nanda @sonadztux

**Review by sonadztux — APPROVED** [review]
Gue refactor cukup banyak:
- Endpoint GET `/face-payment/account/:session_id` buat cek apakah suatu session_id udah punya akun aktif atau belum.
- Endpoint POST `/check-limit` buat cek limit minimum payment sama dapetin value `full_name`, `balance`, `have_twin`. Kalo `is_limit = true` maka perlu pin checking.
- Refactor fungsi di model face payment: ActivateAccount & CreateAccountWallet

Minta tolong hasil refactor dari gue dicoba dulu sebelum merge.

Create transaction gue coba udah aman, tapi @bagasteguhimani atau @lazuardyk dokumentasinya belum sesuai sample request payload-nya, minta tolong disesuain yak. Thanks.

**Pull request merged**